canopen_ros2_controllers/test/test_canopen_proxy_controller.cpp

```diff
@@ -0,0 +1,280 @@
+{
+  SetUpController(false);
+
+  ASSERT_TRUE(controller_->joint_name_.empty());
+
+  ASSERT_EQ(controller_->on_configure(rclcpp_lifecycle::State()), NODE_ERROR);
+
+  ASSERT_TRUE(controller_->joint_name_.empty());
+}
+
+TEST_F(CanopenProxyControllerTest, all_parameters_set_configure_success)
+{
+  SetUpController();
+
+  ASSERT_TRUE(controller_->joint_name_.empty());
+
+  ASSERT_EQ(controller_->on_configure(rclcpp_lifecycle::State()), NODE_SUCCESS);
+
+  // TODO(livanov93): check if it is string
+  // hardcoded as it should be "joint1"
+  ASSERT_THAT(controller_->joint_name_, testing::HasSubstr("joint"));
+}
+
+TEST_F(CanopenProxyControllerTest, check_exported_intefaces)
+{
+  SetUpController();
+
+  ASSERT_EQ(controller_->on_configure(rclcpp_lifecycle::State()), NODE_SUCCESS);
+
+  auto command_intefaces = controller_->command_interface_configuration();
+  ASSERT_EQ(command_intefaces.names.size(), joint_command_values_.size());
+  for (size_t i = 0; i < command_intefaces.names.size(); ++i) {
+    EXPECT_EQ(command_intefaces.names[i], joint_name_ + "/" + command_interface_names_[i]);
+  }
+
+  auto state_intefaces = controller_->state_interface_configuration();
+  ASSERT_EQ(state_intefaces.names.size(), joint_state_values_.size());
+  for (size_t i = 0; i < state_intefaces.names.size(); ++i) {
+    EXPECT_EQ(state_intefaces.names[i], joint_name_ + "/" + state_interface_names_[i]);
+  }
+}
+
+TEST_F(CanopenProxyControllerTest, activate_success)
+{
+  SetUpController();
+
+  ASSERT_EQ(controller_->on_configure(rclcpp_lifecycle::State()), NODE_SUCCESS);
+  ASSERT_EQ(controller_->on_activate(rclcpp_lifecycle::State()), NODE_SUCCESS);
+
+  // check that the message is reset
+  auto msg = controller_->input_cmd_.readFromNonRT();
+  EXPECT_EQ((*msg)->index, 0u);
+  EXPECT_EQ((*msg)->subindex, 0u);
+  EXPECT_EQ((*msg)->type, 0u);
+  EXPECT_EQ((*msg)->data, 0u);
+
+}
+
+TEST_F(CanopenProxyControllerTest, update_success)
+{
+  SetUpController();
+
+  ASSERT_EQ(controller_->on_configure(rclcpp_lifecycle::State()), NODE_SUCCESS);
+  ASSERT_EQ(controller_->on_activate(rclcpp_lifecycle::State()), NODE_SUCCESS);
+
+  ASSERT_EQ(
+    controller_->update(rclcpp::Time(0), rclcpp::Duration::from_seconds(0.01)),
+    controller_interface::return_type::OK);
+}
+
+TEST_F(CanopenProxyControllerTest, deactivate_success)
+{
+  SetUpController();
+
+  ASSERT_EQ(controller_->on_configure(rclcpp_lifecycle::State()), NODE_SUCCESS);
+  ASSERT_EQ(controller_->on_activate(rclcpp_lifecycle::State()), NODE_SUCCESS);
+  ASSERT_EQ(controller_->on_deactivate(rclcpp_lifecycle::State()), NODE_SUCCESS);
+}
+
+TEST_F(CanopenProxyControllerTest, reactivate_success)
+{
+  SetUpController();
+
+  ASSERT_EQ(controller_->on_configure(rclcpp_lifecycle::State()), NODE_SUCCESS);
+  ASSERT_EQ(controller_->on_activate(rclcpp_lifecycle::State()), NODE_SUCCESS);
+  ASSERT_EQ(controller_->command_interfaces_[CommandInterfaces::TPDO_DATA].get_value(), 101.101);
```

Comment:
Can we all all interface to test their values? Maybe would also be a good idea to slightly change their value.

Reply:
@destogl I planned to add more attention to the value propagation and canopen logic on 402 profile implementation.